packages/@aws-cdk/aws-elasticsearch/lib/domain.ts

```diff
@@ -0,0 +1,611 @@
+import * as cdk from '@aws-cdk/core';
+import * as iam from '@aws-cdk/aws-iam';
+import * as kms from '@aws-cdk/aws-kms';
+import * as logs from '@aws-cdk/aws-logs';
+import { Metric, MetricOptions, Statistic } from '@aws-cdk/aws-cloudwatch';
+import { EbsDeviceVolumeType } from '@aws-cdk/aws-ec2';
+import { PolicyStatement } from '@aws-cdk/aws-iam';
+import { LogGroupResourcePolicy } from './log-group-resource-policy';
+import { CfnDomain } from './elasticsearch.generated';
+
+export interface ClusterConfig {
+  readonly masterNodes: number;
+  readonly masterNodeInstanceType: string;
+  readonly dataNodes: number;
+  readonly dataNodeInstanceType: string;
```

Comment:
It seems like we can choose some sensible defaults here.

Reply:
The configuration here has impacts on the cost, availability and performance of the cluster so I didn't want to set defaults here.

If we chose sensible defaults it would exceed the free tier, I think.